cli/internal/oauth/manager/util.go

```diff
@@ -0,0 +1,43 @@
+package manager
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docker/cli/cli/config/credentials"
+	"github.com/docker/cli/cli/version"
+	"github.com/shirou/gopsutil/v3/host"
+)
+
+const (
+	audience = "https://hub.docker.com"
+	tenant   = "login.docker.com"
+	clientID = "DHWuMefQ1v4lxENpz8oUYH50yYSwyPvi"
+)
+
+func NewManager(store credentials.Store) (*OAuthManager, error) {
+	hostinfo, err := host.Info()
+	if err != nil {
+		return nil, err
+	}
+
+	cliVersion := strings.ReplaceAll(version.Version, ".", "_")
+	options := OAuthManagerOptions{
+		Audience:   audience,
+		ClientID:   clientID,
+		Tenant:     tenant,
+		DeviceName: "docker-cli:" + cliVersion,
+		Store:      store,
+	}
+
+	if hostinfo != nil {
+		hostVersion := strings.ReplaceAll(hostinfo.PlatformVersion, ".", "_")
+		options.DeviceName = fmt.Sprintf("docker-cli:%s:%s-%s-%s", cliVersion, hostinfo.OS, hostVersion, hostinfo.KernelArch)
+	}
```

Comment:
So I went looking "What information do we _really_ need here?"

- `cliVersion` - :warning: this function is set at compile time; there is no formal definition of the format (and can be set to anything)
- `hostinfo.OS` - this is hardcoded to `runtime.GOOS` in [`gopsutil/host.Info()`](https://github.com/shirou/gopsutil/blob/865b8c3f58d6ee4c47dab074fbf3c4b037abfd96/host/host.go#L65-L69)
- `hostVersion` - this comes from `PlatformVersion`. Is it a strong requirement to know the version of the OS? On Linux it will be useless without distro. On Windows, it will be "guesswork" (no official API); if _really_ needed, we could also use [`pkg/parsers/kernel.GetKernelVersion()`](https://github.com/moby/moby/blob/1205a9073320fba9e67fa5de3857f0330e56ce50/pkg/parsers/kernel/kernel_darwin.go#L13-L24) from the docker code.
- `hostinfo.KernelArch` - do we need detailed architecture (including CPU sub-variants), which could even be "off" when running with QEMU emulation; could we use `runtime.GOARCH` instead?


Can we get rid of this dependency, and replace `host.Info()` with something more basic?